src/frontend/js/index.js

```diff
@@ -96,7 +98,17 @@ const nerf = x => (1+erf(x*2-1))/2
 
 // get data from Julia:
 const getData = async endpoint =>
-    (await fetch("/api/" + endpoint)).json().catch(e => console.error(`Error getting data from:\n/api/${endpoint}\n\n`, e))
+    (await fetch("/api/" + endpoint + (endpoint.includes("?") ? "&" : "?") + "v=" + (await getApiVersion()))).json().catch(e => console.error(`Error getting data from:\n/api/${endpoint}\n\n`, e))
+
+async function _getApiVersion() {
+    try {
+        return (await (await fetch("/api/version")).json())["version"]
+    } catch (e) {
+        console.error(`Error getting data from:\n/api/version\n\n`, e)
+    }
+}
+
+const getApiVersion = memoize(_getApiVersion)
```

Comment:
Seems a bit silly to use memoize for a function with no args?

`const API_VERSION_PROMISE = getApiVersion()` would be okay, I think?

Reply:
Yes, it is silly. Ta